crates/runtime/src/dataconnector/flight.rs

```diff
@@ -57,3 +58,30 @@ impl Flight {
         })
     }
 }
+
+// Render a user-friendly message for when a FlightError occurs.
+pub fn render_flight_error(error: FlightError, ctx: &str) -> String {
```

Comment:
We shouldn't include details about the full error in the main logs - that should go behind a debug! trace and the user-friendly error is shown in error! \

See the guide to user facing errors: https://github.com/spiceai/spiceai/blob/trunk/docs/dev/user_facing_errors.md